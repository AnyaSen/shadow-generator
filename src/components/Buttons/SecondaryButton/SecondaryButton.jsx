import React from "react";

import Styles from "./SecondaryButton.module.scss";

export default function SecondaryButton({ text }) {
  return <button className={Styles.Button}>{text}</button>;
}
