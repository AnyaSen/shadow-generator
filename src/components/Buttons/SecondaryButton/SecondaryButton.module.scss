@import "../../../assets/scss/mixins";
@import "../../../assets/scss/variables";

.Button {
  @include button($color-grey, $color-dark-grey, $color-text-color);
  width: fit-content;
}
