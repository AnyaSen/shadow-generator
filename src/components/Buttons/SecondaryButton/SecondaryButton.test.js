import React from "react";
import { render, cleanup } from "@testing-library/react";

import SecondaryButton from "./SecondaryButton";

describe("<SecondaryButton />", () => {
  afterEach(cleanup);

  test("Should have correct text.", () => {
    const text = "Go Back";

    const { getByText, container } = render(<SecondaryButton text={text} />);

    const button = getByText(text);

    expect(button).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
