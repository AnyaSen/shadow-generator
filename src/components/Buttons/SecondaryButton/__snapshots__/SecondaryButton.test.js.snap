// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecondaryButton /> Should have correct text. 1`] = `
<button
  class="Button"
>
  Go Back
</button>
`;
